### PR TITLE
[translation] Spanish (es-SV)

### PR DESCRIPTION
**Pull request by ricardoerl**
### User Interface

- [x] General
- [x] Opinions
- [x] Generic Opinions
- [x] Salary Ranges
- [x] Company Size
- [x] Years of Experience
- [x] Sections
- [x] Tools
- [x] Pages
- [x] Blocks
- [x] Reasons
- [x] Quadrant
- [x] Awards
- [x] Sharing

### Introductions

- [x] Awards
- [x] Back-end Frameworks
- [x] Connections
- [x] Data Layer
- [x] Demographics
- [x] Front-end Frameworks
- [x] Introduction
- [x] Flavors
- [x] Mobile & Desktop
- [x] Opinions
- [x] Other Tools
- [x] Support
- [x] Testing

### Conclusions

- [x] Back-end Frameworks
- [x] Conclusion
- [x] Data Layer
- [x] Front-end Frameworks
- [x] Flavors
- [x] Mobile & Desktop
- [x] Testing

### CODE REVIEW

**Comment by joelalejandro** [comment]
@ricardoerl Hey, was about to jump in and get started translating but saw this PR first. How do you feel about using the generic `es` locale to cover all Spanish-speaking countries in Central & South America? That way you could leave `es-ES` for Spain.

File: surveys/2018/website/src/translations/es-SV.yml [hunk 1]
```diff
@@ -0,0 +1,752 @@
...
+    - key: percentage_of_respondents
+      t: ${percentage} de los encuestados
+    - key: partners.premium_partners
+      t: Nuestros Socios Premium 2018
+    - key: partners.become_partner
+      t: Conviértete en un socio
+    - key: partners.thanks
+      t: |
+        Gracias a nuestros socios por apoyar este proyecto.
+        <a href="/support">Aprender más</a> sobre cómo apoyar The State of JS.
```
Comment (joelalejandro) [inline surveys/2018/website/src/translations/es-SV.yml]: @ricardoerl Here we could use the imperative "Aprende" and add the "a" preposition before "The State of JS":

```
Aprende más sobre cómo apoyar a The State of JS
```

**Comment by ricardoerl** [comment]
@joelalejandro yeah, let's use the generic. I just thought that sometimes the differences between Latin American spanish and spanish from Spain are very noticeable, so maybe leave room for both. However I don't have any problem in using the generic lang code.

Thanks for the help :+1:  I already added you as a collaborator.

File: surveys/2018/website/src/translations/es-SV.yml (hunk 1 above)
Comment (ricardoerl) [reply surveys/2018/website/src/translations/es-SV.yml]: That sounds way better, thanks. I'll update that phrase.

**Comment by joelalejandro** [comment]
@ricardoerl Just pushed the entire "User Interface" section. In the interest of making the text more inclusive, I've used "personas" instead of "desarrolladores" for the term "developers", in an attempt to keep it gender-neutral. I'm not entirely sure if that's the best way to go about it. Any feedback is appreciated :)

**Comment by joelalejandro** [comment]
Updated status:

> ### User Interface
> * [x]  General
> * [x]  Opinions
> * [x]  Generic Opinions
> * [x]  Salary Ranges
> * [x]  Company Size
> * [x]  Years of Experience
> * [x]  Sections
> * [x]  Tools
> * [x]  Pages
> * [x]  Blocks
> * [x]  Reasons
> * [x]  Quadrant
> * [x]  Awards
> * [x]  Sharing
> 
> ### Introductions
> * [ ]  Awards
> * [ ]  Back-end Frameworks
> * [ ]  Connections
> * [ ]  Data Layer
> * [ ]  Demographics
> * [ ]  Front-end Frameworks
> * [ ]  Introduction
> * [ ]  Flavors
> * [ ]  Mobile & Desktop
> * [ ]  Opinions
> * [ ]  Other Tools
> * [ ]  Testing
> 
> ### Conclusions
> * [ ]  Back-end Frameworks
> * [ ]  Conclusion
> * [ ]  Data Layer
> * [ ]  Front-end Frameworks
> * [ ]  Flavors
> * [ ]  Mobile & Desktop
> * [ ]  Testing

**Comment by ricardoerl** [comment]
@joelalejandro Did some progress with the "Introductions" section and updated the description of the pr to reflect the changes.

I was also considering changing the pr locale to `es-ES` (branch name and pr title) but this would require deleting the current pr in order to create a new one with the updated information. I guess this is not a blocker but should we do this now or can we wait until all the translation work is completed?

Happy holidays!

File: surveys/2018/website/src/translations/es-SV/introductions/awards-introduction.md [hunk 2]
```diff
@@ -3,5 +3,5 @@ type: introduction
 section: awards
 locale: es-SV
 ---
- Join us for the first ever State of JS awards! Can you guess which technologies took gold
-in each category? <span class="note">(Note: we've excluded ES6 from the awards to avoid skewing the data)</span>
+ ¡Unete a nosotros para los primeros premios de State of JS!
```
Comment (joelalejandro) [inline surveys/2018/website/src/translations/es-SV/introductions/awards-introduction.md]: It should say "Únete"

File: surveys/2018/website/src/translations/es-SV/introductions/connections-introduction.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+type: introduction
+section: connections
+locale: es-SV
+---
+¿Cuántos usuarios de *React* también usan *Redux*? Personas aficionadaas *GraphQL* prefieren *Jest*?
+¿Hay usuarios de *Express* también con interés por *Ember*?
+
+El tamaño de cada sección corresponde al número de personas encuestadas que han utilizado cada biblioteca, y que estarían dispuestas a usarla nuevamente.
```
Comment (joelalejandro) [inline surveys/2018/website/src/translations/es-SV/introductions/connections-introduction.md]: Probably best to use "librería"? I think that term is more familiar.

**Comment by joelalejandro** [comment]
@ricardoerl Let's hold on until we're finished. In the meantime we can review each other's translation work here; the final PR will look more polished this way :)

**Comment by SachaG** [comment]
What's the status on the Spanish version? I would love to have it available when we "officially" announce the translated versions :)

**Comment by joelalejandro** [comment]
@SachaG I can get it finished in a couple of days :) Been out of Git recently but will finish this very soon!

**Comment by SachaG** [comment]
No worries, thanks!

**Comment by joelalejandro** [comment]
@SachaG How do you feel about getting a more neutral text for the testing introduction in other languages?

The English version says:

> Testing, testing… Is this thing on? Sorry, I Jest, but I would never want to make a Mocha-ry 
of this section. That would be Jasmine (“just mean”? get it?), and that's bad Karma.

While it's quite clever and funny, it can't be translated into Spanish (or any other languages, unless they have vocabulary that can mimic the phonetic gist of the joke).

**Comment by joelalejandro** [comment]
Also @ricardoerl, feel free to mark the following introductions as translated in the PR description:

- Data Layer
- Demographics
- Front-end Frameworks
- Introduction
- Flavors
- Mobile & Desktop
- Opinions
- Other Tools

And also, would you please add a tick for `Support` (which wasn't in the original list)? Thanks!

**Comment by SachaG** [comment]
> While it's quite clever and funny, it can't be translated into Spanish (or any other languages, unless they have vocabulary that can mimic the phonetic gist of the joke).

Yes, feel free to write your own into, you don't have to try and translate that same joke.

**Comment by ricardoerl** [comment]
PR progress updated.

File: surveys/2018/website/src/translations/es-SV/conclusions/back-end-frameworks-conclusion.md [hunk 4]
```diff
@@ -3,8 +3,8 @@ type: conclusion
...
+ JavaScript en el servidor está en una situación extraña. Aunque cada año surgen innumerables frameworks muy pocos logran ganar el impulso suficiente para desafiar a **Express**. Incluso **Koa**, a veces considerado como el sucesor de Express, tiene un índice de satisfacción bajo (y números de uso mucho más bajos).
 
-One interesting entrant in the space is **Next.js**, which has been generating a lot of interest lately. Although it's not quite comparable to a full-featured Node back-end, its single-minded focus on solving the server-side-rendering problem for React apps has made it a very useful tool. 
+Un interesante nuevo participante en el espacio es **Next.js**, el cual ha generado mucho interés últimamente. Aunque no es del todo comparable a un back-end completo de Node, su enfoque exclusivo en la solución del problema de *server-side-rendering* para aplicaciones en React lo ha convertido en una herramienta muy útil.
```
Comment (joelalejandro) [inline surveys/2018/website/src/translations/es-SV/conclusions/back-end-frameworks-conclusion.md]: @ricardoerl Maybe we can replace "interesante" here with a synonym like "llamativo", so we don't repeat ourselves in the next paragraph?

**Comment by joelalejandro** [comment]
@SachaG I've translated the testing introduction with the following text:

> Las aplicaciones web se están volviendo cada vez más complejas, con más y más lógica implementada en el cliente. Es de vital importancia testear todo este código que escribimos y, afortunadamente, el testing es una de las áreas que más diversidad de soluciones ofrece.

In English:

> Web applications are becoming more and more complex, with even more logic being written client-side. It is vital to test all of the code we write and, luckily, testing is one of the areas with the most variety of solutions to offer.

**Comment by joelalejandro** [comment]
**Translation has been completed!**

@ricardoerl Feel free to review and if everything looks good, we can create the proper PR/branch to use the lang code `es`.

**Comment by SachaG** [comment]
Congrats! I'm excited to merge this :)

**Comment by joelalejandro** [comment]
@ricardoerl Everything looks good. Feel free to close this PR.

**Pull request closed without merging**